Libraries/Opc.Ua.Client/NodeCache.cs

```diff
@@ -621,10 +748,51 @@ public void FetchSuperTypes(ExpandedNodeId nodeId)
                 includeSubtypes,
                 m_typeTree);
 
-            foreach (IReference reference in references)
+            var targetIds = new ExpandedNodeIdCollection(
+                references.Select(reference => reference.TargetId));
+
+            IList<INode> result = Find(targetIds);
+
+            foreach (INode target in result)
             {
-                INode target = Find(reference.TargetId);
+                if (target != null)
+                {
+                    targets.Add(target);
+                }
+            }
+            return targets;
+        }
+
```

Comment:
Related to my previous comment (*1) the implementation of FindReferences that receives a list of ReferenceTypeIds would add to the targetIds collection all requested referenceTypeIds:
```
foreach (var source in sources)
{
    if (!(source is Node node))
    {
        continue;
    }

    foreach (var referenceTypeId in referenceTypeIds)
    {
        IList<IReference> references = node.ReferenceTable.Find(
            referenceTypeId,
            isInverse,
            includeSubtypes,
            m_typeTree);

        targetIds.AddRange(
            references.Select(reference => reference.TargetId));
    }
}
```

and then, the method that requests only one ReferenceTypeId can call the new FindReferences method passing an array with a single element.

Reply:
added